plugins/parsers/csv/parser.go

```diff
@@ -117,15 +199,28 @@ func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
 }
 
 func parseCSV(p *Parser, r io.Reader) ([]telegraf.Metric, error) {
-	csvReader := p.compile(r)
+	lineReader := bufio.NewReader(r)
 	// skip first rows
 	for p.SkipRows > 0 {
-		_, err := csvReader.Read()
-		if err != nil {
+		line, err := lineReader.ReadString('\n')
+		if err != nil && len(line) == 0 {
 			return nil, err
 		}
 		p.SkipRows--
```

Comment:
I think this is wrong now as it does not decrement the `SkipRows` count if the line is empty.
```suggestion
		_, err := lineReader.ReadString('\n')
		if err != nil {
			return nil, err
		}
		p.SkipRows--
```

Reply:
Could even be like this?
```suggestion
		if _, err := lineReader.ReadString('\n'); err != nil {
			return nil, err
		}
		p.SkipRows--
```

Reply:
@etycomputer What do you think?

Reply:
This code is still wrong. Imagine you have the following data
```csv
# My skip row stuff

yeah it even might contain totally empty lines




wooohhhaaa
time,value,tag
```
then all those empty lines (only containing `\n`) will return `nil, nil` in your code but **NOT** decrement `SkipRows`!

Reply:
Even an empty line will contain the `\n` and thus has a length greater than zero. So the above code will work.